src/gg.regression.unity.bots/Runtime/Scripts/Validation/RGValidateBehaviour.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using StateRecorder.BotSegments.Models.SegmentValidations;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace RegressionGames.Validation
+{
+    public class RGValidateBehaviour: MonoBehaviour
```

Comment:
We should model this differently.  Instead of spawning monobehaviours and running into timing issues where endCriteria match before we've even spawned these, why not handle instantiation/evaluation as part of our botsegment evaluation loop like actions/endConditions are done ?

So instead of these being behaviours that self process on the Update loop, they would be implementations of an interface that a ValidationEvaluator behaviour driven by our botsegment update loop would process through on each update pass.

This would remove your ambiguous timing issues.

The one user code difference is that these wouldn't be MonoBehaviour objects anymore, but given that we aren't parenting these to anything useful anyway, I see that as an improvement.  The APIs the user needs to find objects are still available to them MonoBehaviour or not because we'll be calling them from the main thread Update loop.